tcf_website/management/commands/load_grades.py

```diff
@@ -118,159 +179,135 @@ def load_row_into_dict(self, row):
 
-        # value of dictionaries (incremented onto value if key already exists)
+        # Dictionary values (incremented onto value if key already exists)
         this_semesters_grades = [a_plus, a, a_minus,
                                  b_plus, b, b_minus,
                                  c_plus, c, c_minus,
-                                 d_plus, d, d_minus,
-                                 f,
-                                 other, drop, withdraw]
-
-        # load this semester into course dictionary
-        if course_identifier in course_grades:
-            for i in range(len(course_grades[course_identifier])):
-                course_grades[course_identifier][i] += this_semesters_grades[i]
+                                 dfw]
+
+        # Load this semester into course_grades dictionary
+        if course_identifier in self.course_grades:
+            for i in range(len(self.course_grades[course_identifier])):
+                self.course_grades[course_identifier][i] += this_semesters_grades[i]
         else:
-            course_grades[course_identifier] = this_semesters_grades
+            self.course_grades[course_identifier] = this_semesters_grades.copy()
 
-        # load this semester into course instructor dictionary
-        if course_instructor_identifier in course_instructor_grades:
-            for i in range(
-                    len(course_instructor_grades[course_instructor_identifier])):
-                course_instructor_grades[course_instructor_identifier][i] += this_semesters_grades[i]
+        # Load this semester into course_instructor_grades dictionary
+        if course_instructor_identifier in self.course_instructor_grades:
+            for i in range(len(self.course_instructor_grades[course_instructor_identifier])):
+                self.course_instructor_grades[course_instructor_identifier][i] += this_semesters_grades[i]
         else:
-            course_instructor_grades[course_instructor_identifier] = this_semesters_grades
+            self.course_instructor_grades[course_instructor_identifier] = this_semesters_grades.copy(
+            )
 
     def load_dict_into_models(self):
+        """ Converts dictionaries to real instances of CourseGrade and CourseInstructorGrade.
+
+        Given a course or course-instructor pair and its corresponding grade distribution,
+        calculates weighted GPA average and total enrolled students and then uses all of those
+        as parameters to create new CourseGrade and CourseInstructorGrade instances.
+        """
         if self.verbosity > 0:
             print('Step 2: Bulk-create CourseGrade instances')
-        # used for gpa calculation
+        # Used for GPA calculation
         grade_weights = [4.0, 4.0, 3.7,
                          3.3, 3.0, 2.7,
                          2.3, 2.0, 1.7,
-                         1.3, 1.0, 0.7,
-                         0,
-                         0, 0, 0]
+                         0]
```

Comment:
Are grade calculations are changed by a large amount based on counting dfw as 0? Slightly worried if a class has a sizable amount of withdraws this will impact the average significantly.

Reply:
nvm I see that later on in the helper function you remove the DFW from the denominator. Maybe make a comment about it? also fine as is.